lib/change_stream.js

```diff
@@ -142,22 +150,28 @@ class ChangeStream extends EventEmitter {
    */
   next(callback) {
     return maybePromise(this.parent, callback, cb => {
-      if (this.isClosed()) {
-        return cb(new MongoError('ChangeStream is closed'));
-      }
-      this.cursor.next((error, change) => {
-        processNewChange({ changeStream: this, error, change, callback: cb });
+      getCursor(this, (err, cursor) => {
+        if (err) return cb(err); // failed to resume, raise an error
+        cursor.next((error, change) => {
+          if (error) {
+            this[kResumeQueue].push(() => this.next(cb));
+            processError(this, error, cb);
+            return;
+          }
+          processNewChange(this, change, cb);
+        });
       });
     });
   }
 
   /**
-   * Is the cursor closed
+   * Is the change stream closed
    * @method ChangeStream.prototype.isClosed
+   * @param  {boolean} [checkCursor=true] also check if the underlying cursor is closed
    * @return {boolean}
    */
-  isClosed() {
-    return this.closed || (this.cursor && this.cursor.isClosed());
+  isClosed(checkCursor) {
```

Comment:
I'm not sure we should make this change, `isClosed` is here for duck-typed parity with [Cursor#isClosed](https://github.com/mongodb/node-mongodb-native/blob/master/lib/cursor/cursor.js#L1002). Is there a way to implement the behavior that requires this boolean internally?

Reply:
I made this change in response to [this discussion](https://github.com/mongodb/node-mongodb-native/pull/2360#discussion_r423053649). Should I revert it to the original implementation?

Reply:
Yeah I think I was mostly asking for rationale in that comment, and check that you knew you were doing that or if there was a bug. I think not exposing this at the api level is the right thing to do (users don't want to check isClosed conditionally, they just want to know if its closed)